src/app/WriteHandler.cpp

```diff
@@ -788,9 +788,13 @@ DataModel::ActionReturnStatus WriteHandler::CheckWriteAllowed(const Access::Subj
     {
         // NOTE: explicit cast/check only for attribute path and nothing else.
         //
-        //       In particular `request.path` is a DATA path (contains a list index)
-        //       and we do not want mLastSuccessfullyWrittenPath to be auto-cast to a
+        //       Specifically we DO NOT use `operator==` because `aPath` is a
+        //       DATA path (contains a list index) and we do not want
```

Comment:
Again: why is aPath a data path?  It should be a `const ConcreteAttributePath &` because we in fact do not care about the list bits in this code.